### ISSUE TITLE
GitHub teams references in the approved-reviews-by condition

### ISSUE THREAD

**Issue by bowlofeggs** [Mergifyio/mergify-engine#310]
Greetings!

I am attempting to configure my project with a rule that matches on the @fedora-infra/bodhi team as a reviewer, like this:

```
- approved-reviews-by=[@fedora-infra/bodhi]
```

That is based on my understanding of the [docs](https://doc.mergify.io/conditions.html#attribute-list), which at the time of this writing say "approved-reviews-by | array of string | The list of GitHub user or team login that approved the pull request. Team logins are prefixed with the @ character. This will match only reviewers with admin or write permission on the repository."

When I try this out, the mergify config checker fails and complains about the use of the @ character:

```
Mergify configuration is invalid: extra keys not allowed @ data['pull_request_rules'], Invalid condition 'approved-reviews-by=[@fedora-infra/bodhi]'. Expected end of text (at char 34), (line:1, col:35) @ data['pull_request_rules'][1]['conditions'][2]
```

Am I formatting it incorrectly?

**Comment by bowlofeggs** [comment]
I did try @bodhi and the config checker seems to be OK with it (it's when the / is present that it gets upset it seems, despite the error about @), but then it doesn't seem to match and actually merge PRs when they are reviewed by members of the bodhi team.

**Comment by bowlofeggs** [comment]
I am also having trouble with another rule with this match line:

```
- approved-reviews-by=[bowlofeggs]
```

A pull request I've reviewed is not getting merged, and Mergify says that this rule isn't satisfied. The docs say this can be an "array of string" - is the syntax I used for an array of strings not correct here?

**Comment by jd** [comment]
You should just use `- approved-reviews-by=bowlofeggs`. If the value is an array of string, the `=` operator acts as a `in` operator.

**Comment by bowlofeggs** [comment]
What is the syntax for expressing an array of strings?

**Comment by bowlofeggs** [comment]
I tried ```- approved-reviews-by=@fedora-infra/bodhi``` and Mergify says the [syntax is invalid](https://github.com/fedora-infra/bodhi/pull/2814/checks?check_run_id=37944764):

```
Mergify configuration is invalid: extra keys not allowed @ data['pull_request_rules'], Invalid condition 'approved-reviews-by=@fedora-infra/bodhi'. Expected end of text (at char 33), (line:1, col:34) @ data['pull_request_rules'][1]['conditions'][2]
```

**Comment by jd** [comment]
The `/` character is not allowed here because we expect a GitHub login. We probably need to allow it since teams can have it.

**Comment by jd** [comment]
So here's a few things:

- Team are actually not supported in `approved-reviews-by`. The doc is wrong about that. The only field that supports team is `review-requested`. That's what GitHub supports. I've fixed the doc in #314.
- The parsing for `review-requested` with a team that starts with `@` was broken. Fixed in #313.
- Having team supported in `approved-reviews-by` and similar field is a feature request. I've used this issue as a feature request for it and changed the title/tag.
